core/domain/subtopic_page_domain.py

```diff
@@ -92,25 +92,107 @@ def to_dict(self):
         return subtopic_page_change_dict
 
 
+class SubtopicPageContents(object):
+    """Domain object for the contents on a subtopic page."""
+
+    def __init__(
+            self, subtitled_html, content_ids_to_audio_translations):
+        """Constructs a SubtopicPageContents domain object.
+
+        Args:
+            subtitled_html: SubtitledHtml. The html data being displayed on
+                the page.
+            content_ids_to_audio_translations: dict. The audio translations
+                that are a part of the subtopic page, organized by language
+                code.
+        """
+        self.subtitled_html = subtitled_html
+        self.content_ids_to_audio_translations = (
+            content_ids_to_audio_translations)
+
+    def validate(self):
+        self.subtitled_html.validate()
+
+        # TODO(tjiang11): Extract content ids to audio translations out into
+        # its own object to reuse throughout audio-capable structures.
+        if not isinstance(self.content_ids_to_audio_translations, dict):
+            raise utils.ValidationError(
+                'Expected content_ids_to_audio_translations to be a dict,'
+                'received %s' % self.content_ids_to_audio_translations)
+        for (content_id, audio_translations) in (
+                self.content_ids_to_audio_translations.iteritems()):
+
+            if not isinstance(content_id, basestring):
+                raise utils.ValidationError(
+                    'Expected content_id to be a string, received: %s' %
+                    content_id)
+            if not isinstance(audio_translations, dict):
+                raise utils.ValidationError(
+                    'Expected audio_translations to be a dict, received %s'
+                    % audio_translations)
+
+            allowed_audio_language_codes = [
+                language['id'] for language in (
+                    constants.SUPPORTED_AUDIO_LANGUAGES)]
+            for language_code, translation in audio_translations.iteritems():
+                if not isinstance(language_code, basestring):
+                    raise utils.ValidationError(
+                        'Expected language code to be a string, received: %s' %
+                        language_code)
+
+                if language_code not in allowed_audio_language_codes:
+                    raise utils.ValidationError(
+                        'Unrecognized language code: %s' % language_code)
+
```

Comment:
Ditto, can we test whether it contains all of the content_ids?

Reply:
done